cyipopt/tests/unit/test_scipy_optional.py

```diff
@@ -70,3 +70,49 @@ def test_minimize_ipopt_nojac_constraints_if_scipy():
     expected_res = np.array([1.001867, 0.99434067, 1.05070075, 1.17906312,
                              1.38103001])
     np.testing.assert_allclose(res.get("x"), expected_res)
+
+
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
+def test_minimize_ipopt_jac_and_hessians_constraints_if_scipy():
+    """ `minimize_ipopt` works with objective gradient and Hessian 
+         and constraint jacobians and Hessians"""
+    from scipy.optimize import rosen, rosen_der, rosen_hess
+    x0 = [1.3, 0.7, 0.8, 1.9, 1.2]
+    constr = {"fun": lambda x: rosen(x) - 1.0, "type": "ineq",
+              'jac': rosen_der, 'hess': lambda x, v: rosen_hess(x) * v[0]}
```

Comment:
Please keep code style consistent within files.
```suggestion
              "jac": rosen_der, "hess": lambda x, v: rosen_hess(x) * v[0]}
```